tools/run_tests/run_xds_tests.py

```diff
@@ -950,6 +969,289 @@ def prepare_services_for_urlmap_tests(gcp, original_backend_service,
     return original_backend_instances, alternate_backend_instances
 
 
+def test_metadata_filter(gcp, original_backend_service, instance_group,
+                         alternate_backend_service, same_zone_instance_group):
+    logger.info("Running test_metadata_filter")
+    wait_for_healthy_backends(gcp, original_backend_service, instance_group)
+    original_backend_instances = get_instance_names(gcp, instance_group)
+    alternate_backend_instances = get_instance_names(gcp,
+                                                     same_zone_instance_group)
+    patch_backend_service(gcp, alternate_backend_service,
+                          [same_zone_instance_group])
+    wait_for_healthy_backends(gcp, alternate_backend_service,
+                              same_zone_instance_group)
+    try:
+        with open(bootstrap_path) as f:
+            md = json.load(f)['node']['metadata']
+            match_labels = []
+            for k,v in md.items():
+                match_labels.append({'name': k,
+                                     'value': v})
+
+        not_match_labels = [{'name': 'fake',
+                             'value': 'fail'}]
+        test_route_rules = [
```

Comment:
As commented in the test description doc: it is worth adding a test where filter matches both services (either due to MATCH_ANY or MATCH_ALL). Is it an error or  will traffic go to both services (with equal distribution)?

Reply:
hm I believe it only goes the first matched service, but let me verify, will update.

Reply:
ok

Reply:
ok

Reply:
Indeed, I've verified that multiple route rules each matches the metadata filter are allowed, and the first match route rule is selected for all the traffic.

Reply:
I overlooked the priority field in my initial review so my initial comment doesn't make sense. But with the new addition in the latest commit - after applying the new route rule why does MATCH_ANY for the same set of match_labels doesn't match but MATCH_ALL for the same set of labels applies? That's difficult to understand.

Reply:
> after applying the new route rule why does MATCH_ANY for the same set of match_labels doesn't match but MATCH_ALL for the same set of labels applies?

Both of them match. Just the `MATCH_ANY` comes first so that one is selected, which is configured to `alternate_backend_service`, and the traffic switched to there.

Reply:
ok